File: js/110_BalancedBinaryTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {
    if(root === null) return true;
    return isBalanced(root.left) && isBalanced(root.right) && Math.abs(getHeight(root.right) - getHeight(root.left)) <= 1;
};

var getHeight = function(root){
    if(root === null) return 0;
    return Math.max(getHeight(root.left), getHeight(root.right)) + 1;
}

File: cpp/110_BalancedBinaryTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isBalanced(TreeNode* root) {
        int height;
        return isBalanceWithHeight(root,height);
    }
private:
    bool isBalanceWithHeight(TreeNode *root, int& height){
        if(root == NULL){
            height = 0;
            return true;
        }
        int hLeft;
        int hRight;
        if(isBalanceWithHeight(root->left, hLeft) && isBalanceWithHeight(root->right, hRight) && abs(hLeft - hRight) <= 1 ){
            height = max(hLeft, hRight) + 1;
            return true;
        }
        height = max(hLeft, hRight) + 1;
        return false;
    }
};